llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp

```diff
@@ -94,11 +94,60 @@ struct RISCVOutgoingValueHandler : public CallLowering::OutgoingValueHandler {
 
   void assignValueToReg(Register ValVReg, Register PhysReg,
                         CCValAssign VA) override {
+    // If we're passing an f32 value into an i64, anyextend before copying.
+    if (VA.getLocVT() == MVT::i64 && VA.getValVT() == MVT::f32)
```

Comment:
It looks like maybe this bit of code could be handled by setting LocInfo to  CCValAssign::AExt, and letting it fallthrough instead?  Not sure about what else that means though, so take with bucket of salt.

Reply:
I think so too, but it would require moving some code in SelectionDAG. Maybe I'll do it as a follow up.